### PR TITLE
fix: backport libuv patch for failure to load advapi32

### LINKED ISSUE #23841: Electron 9 crashes on boot on Windows 10 2004
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:** 9.0.0
* **Operating System:** Windows 10 Version 2004 (OS Build 19635.1) x64
* **Last Known Working Electron version:** 8.3.0

### Expected Behavior
Electron boots correctly.

### Actual Behavior
Electron crashes early in the boot sequence:

```
>	electron.exe!uv_fatal_error(const int errorno, const char * syscall) Line 62	C
 	electron.exe!uv_winapi_init() Line 149	C
 	electron.exe!uv_init() Line 205	C
 	[Inline Frame] electron.exe!uv__once_inner(uv_once_s * guard, void(*)() callback) Line 51	C
 	electron.exe!uv_once(uv_once_s * guard, void(*)() callback) Line 73	C
 	electron.exe!uv_hrtime() Line 458	C
 	[Inline Frame] electron.exe!node::performance::`dynamic initializer for 'timeOrigin'() Line 42	C++
 	electron.exe!_GLOBAL__sub_I_node_perf.cc() Line 0	C++
 	electron.exe!_initterm(void(*)() * first, void(*)() * last) Line 16	C++
```

syscall is "GetModuleHandleA", GetLastError is "The specified module could not be found."

### To Reproduce
Just try to run electron.exe.

### PR DESCRIPTION
**Pull request by felixrieseberg**
#### Description of Change

This fixes #23841.

Cherry picks the upstream libuv fix for this issue: https://github.com/libuv/libuv/commit/335e8a6d128646e5a19d39dfc677f5a5a555f7cc

That patch was verified by folks in libuv land who experienced the issue not experiencing it after the patch.

Verification here: https://github.com/libuv/libuv/issues/2759#issuecomment-606156912

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).
- [x] This is **NOT A BREAKING CHANGE**. Breaking changes may not be merged to master until 11-x-y is branched.

#### Release Notes

Notes: Fixed: On some Windows machines, especially Windows Insider builds, Electron would crash silently during startup.

### CODE REVIEW

File: patches/node/fix_if_not_already_loaded_load_advapi32.patch [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+As of Electron 9, we have a dependency on advapi32.dll. That dll contains
+everything Windows would consider "advanced", like the registry, Windows
+services, user account management, and some other stuff. It's fair that
+Node expects the module to be loaded.
+
+Chrome explicitly delay-loads it, so on some machines, we've seen Electron
+crash hard because the "gimme advapi32" code is running before the dll
+was loaded. It'll be loaded either way, this patch ensures that we do so
+when we haven't already.
```
Comment (nornagon) [inline patches/node/fix_if_not_already_loaded_load_advapi32.patch:14]: Can we upstream this?

Comment (MarshallOfSound) [reply patches/node/fix_if_not_already_loaded_load_advapi32.patch:14]: Unlikely, we could try.  Upstream doesn't need to handle the fact this module is lazy loaded :shrug:

**Comment by deepak1556** [comment]
> As of Electron 9, we have a dependency on advapi32.dll.

@felixrieseberg I am pretty sure we had this dependency before 9 too, `dumpbin /Dependent electron` on Electron 8 shows


```
Image has the following delay load dependencies:

    ADVAPI32.dll
    ole32.dll
    SETUPAPI.dll
    api-ms-win-core-winrt-string-l1-1-0.dll
    api-ms-win-core-winrt-l1-1-0.dll
    CFGMGR32.dll
    POWRPROF.dll
    MF.dll
    MFReadWrite.dll
    MFPlat.DLL
    WINUSB.DLL
    bthprops.cpl
    BluetoothApis.dll
```

Do you know what specifically changed with 9 that causes this crash ?

**Comment by felixrieseberg** [comment]
Nice, thanks for the catch, @deepak1556!

**Pull request merged**